### PR TITLE
chore: Remove lazy person container

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

Lazy Person container is complicated to reason about and we need to continuously keep track to make sure the container is up to date. Instead we want to get closer to the DB and make sure an update doesn't wipe out properties set in parallel (follow-up changes).

Lazy person loading is useful up to 3K out of 23K (12%) https://metrics.posthog.com/d/TUU-rsT4k/tiina-wip?orgId=1&from=now-24h&to=now&viewPanel=10 Which means that up to 12% of the time we might pull the person info from Redis, but note that Redis is backed by PG, so we might still go to PG, plus the more complex merges likely do a bigger portion of PG load. If we want to best optimize PG reads, we'd benefit from reading from Redis in the personProcessingStep too, that said making sure the data is accurate in Redis is not trivial and could cause data trust issues.

## Changes

- Using the Person object instead. We're always fetching the person during the personStep now, previously when the person had been seen by this thread and the event didn't have any person properties set we wouldn't fetch the person info at this point and we would fetch the person from Redis during personCreation (with PG fallback). Hence this PR **will increase number of read requests going to Postgres**. We could alternatively pass around `undefined` values, but that's more error prone. However note that the majority of events are coming from posthog-js and have `$set` on the event.
- ^ means we're not using Redis caching for persons anymore (cache writes will be removed in a follow-up)
- createEvent returns RawClickHouse event now
- if a person gets deleted we'll still use the person on the event that we got earlier from person process step
- personState:
    - pass person around more explicitly
    - more explicit errors for failures - we don't don't say update failed, when the person creation had failed in the past anymore
    - tests (review hiding whitespace 😉  )
        - running less tests, where PoE isn't relevant not running both & including PoE in the test name, so it's easier to know which one failed
        -  used `handleIdentifyOrAlias` and `updateProperties` explicitly rather than `update` to open the way to make things more explicit and easier to reason about with PersonState (e.g. will follow-up to make `updateIsIdentified` explicit)
        - redesigned how races were created (not using lazyperson container for that anymore ... obviously)
        - removed some / added some to improve coverage

### CODE REVIEW

**Comment by hazzadous** [comment]
Looks like there's a test failing in tests/worker/ingestion/event-pipeline/runner.test.ts

tbh I think it's too specific and we should remove it.

File: plugin-server/src/worker/ingestion/event-pipeline/runner.ts [hunk 1]
```diff
@@ -160,7 +156,7 @@ export class EventPipelineRunner {
     }
 
     private async handleError(err: any, currentStepName: string, currentArgs: any, sentToDql: boolean) {
-        const serializedArgs = currentArgs.map((arg: any) => this.serialize(arg))
+        const serializedArgs = currentArgs
         status.error('🔔', 'step_failed', { currentStepName, err })
         Sentry.captureException(err, { extra: { currentStepName, serializedArgs, originalEvent: this.originalEvent } })
         this.hub.statsd?.increment('kafka_queue.event_pipeline.step.error', { step: currentStepName })
```
Comment (hazzadous) [inline plugin-server/src/worker/ingestion/event-pipeline/runner.ts:162]:
```suggestion
        status.error('🔔', 'step_failed', { currentStepName, err })
        Sentry.captureException(err, { extra: { currentStepName, currentArgs, originalEvent: this.originalEvent } })
        this.hub.statsd?.increment('kafka_queue.event_pipeline.step.error', { step: currentStepName })
```

Or something like that

**Review by hazzadous — APPROVED** [review]
Looks good

Something that would be great would be a way to see how the trace changes with these updates, but that's something we can look into later. Just some rough "we did 500 more pg queries in functional tests" or something would be cool. Or maybe some way to benchmark locally with representative data 🤔

File: plugin-server/src/worker/ingestion/person-state.ts [hunk 2]
```diff
@@ -158,10 +147,30 @@ export class PersonState {
...
+            status.error('🚨', 'get_person_failed', { error, teamId: this.teamId, distinctId: this.distinctId })
+            Sentry.captureException(error, {
+                extra: {
+                    teamId: this.teamId,
+                    distinctId: this.distinctId,
+                    timestamp: this.timestamp,
+                    personUuid: this.newUuid,
+                },
+            })
+            throw error
```
Comment (xvello) [inline plugin-server/src/worker/ingestion/person-state.ts:171]: sidetrack: we'll need to try out (on dev or local devenv) what happens when PG does down: do we properly retry here, or still send to DLQ? If the later, that can be fixed in a second PR.

Ideally, `createOrGetPerson` should emit the `DependencyUnavailableError`, but we'll also want to retry here if a race happens, right?

**Review by xvello — APPROVED** [review]
Great work!

File: plugin-server/src/worker/ingestion/person-state.ts (hunk 2 above)
Comment (xvello) [reply plugin-server/src/worker/ingestion/person-state.ts:171]: yep, that's what I thought. We'll improve on another PR, it's not a regression.

File: plugin-server/src/worker/ingestion/event-pipeline/runner.ts (hunk 1 above)
Comment (tiina303) [reply plugin-server/src/worker/ingestion/event-pipeline/runner.ts:162]: I was being lazy :D

**Pull request merged**